### PR TITLE
Fix query exception arguments

### PR DESCRIPTION
**Pull request by RubtsovAV**
The RequestErrorException extends from the \Exception class and first argument can't be the GoogleError, in there must be a string or empty. 

My fix throw new RequestErrorException without arguments.

### CODE REVIEW

**Comment by RubtsovAV** [comment]
This is dependent on another PR https://github.com/serp-spider/core/pull/10 . You need merge them before.

File: src/Parser/AbstractParser.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ public function parse(GoogleDom $googleDom)
 
     protected function createResultSet(GoogleDom $googleDom)
     {
-        $startingAt = $googleDom->getUrl()->getResultsPerPage() * $googleDom->getUrl()->getParamValue('start', 0);
+        $startingAt = (int) $googleDom->getUrl()->getParamValue('start', 0);
         return new IndexedResultSet($startingAt);
     }
```
Comment (gsouf) [inline src/Parser/AbstractParser.php]: That's not desired here

File: src/GoogleClient.php [hunk 2]
```diff
@@ -107,7 +107,7 @@ public function query(GoogleUrlInterface $googleUrl, Proxy $proxy = null, Cookie
                 if ($errorDom->isCaptcha()) {
                     throw new GoogleCaptchaException(new GoogleCaptcha($errorDom));
                 } else {
-                    throw new RequestErrorException($errorDom);
+                    throw new ResponseException($response);
```
Comment (gsouf) [inline src/GoogleClient.php]: That needs more details, until now I wanted to use the ``HttpResponseErrorException`` https://github.com/serp-spider/core/blob/master/src/Exception/RequestError/HttpResponseErrorException.php

But having a trace of the whole response is better, In addtion it should also contain a message readable for humans. For instance "HTtp response was invalid, status code: $statusCode"

**Comment by gsouf** [comment]
Implemented in 6cf4c7b744f2c2475414e554cfa9e5b2670378ea

**Pull request closed without merging**